### ISSUE TITLE
JIT gives wrong gradients

### ISSUE THREAD

**Issue by fehiepsi** [pytorch/pytorch#13669]
## 🐛 Bug

In some situation, JIT gives wrong gradients in backward pass. The following script shows the problem. Note that not only `.exp()` but also with operators such as `.log()`, we get the same issue.

## To Reproduce

```
import torch
import torch.autograd as autograd

def fn(z):
    return (z.exp() - 0).sum() + (z.exp() - torch.zeros(1000)).sum()

z = torch.tensor(1., requires_grad=True)
fn_jit = torch.jit.trace(fn, (z,))
print(fn(z))
print(fn_jit(z))
print(autograd.grad(fn(z), (z,)))
print(autograd.grad(fn_jit(z), (z,)))
```
Output:
```
tensor(2721.0005, grad_fn=<AddBackward0>)
tensor(2721.0005, grad_fn=<AddBackward0>)
(tensor(2721.),)
(tensor(5436.5645),)  # expected 2721
```
## Expected behavior

Gradients of `fn` and `fn_jit` should be equal.

## Environment

- PyTorch Version: nightly build (20181022) using `conda`
- OS: Linux
- Python version: 3.6

**Comment by zou3519** [comment]
This seems like an edge case in autodiff

**Comment by apaszke** [comment]
This is a well known bug that broadcasting in forward causes our gradients to be incorrect (because we don’t reduce over broadcasted dims)

**Comment by fehiepsi** [comment]
@zou3519 Given that this is the only issue left with label 1.0, can this issue be marked by a newer label (1.1 e.g.)?

**Comment by apaszke** [comment]
If this is caused by broadcasting, this should be fixed in 1.0 (I’m not at a computer so I can’t check for sure).

**Comment by fehiepsi** [comment]
Thanks for letting me know! I’ll check it and get back to you.

**Comment by zou3519** [comment]
I checked and it is indeed fixed in the release, thanks to #14485
